src/App.js

```diff
@@ -14,11 +16,120 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export default class App extends React.Component {
+state = {
+  goodsList: [],
+  value: 'Dumplings',
+  hidden: false,
+}
+
+hiddenButton = () => {
+  this.setState({
+    goodsList: [...goodsFromServer],
+    hidden: true,
+  });
+}
+
+reverseButton = () => {
+  this.setState({ goodsList: [...goodsFromServer].reverse() });
+}
+
+sortABCButton = () => {
+  this.setState({ goodsList: [...goodsFromServer].sort() });
+}
+
+resetButton = () => {
+  this.setState({
+    goodsList: [...goodsFromServer],
+    value: 'Dumplings',
+  });
+}
+
+sortByLengthButton = () => {
+  this.setState({ goodsList: [...goodsFromServer]
+    .sort((a, b) => a.length - b.length) });
+}
+
+generateNumForList = (event) => {
+  this.setState({
+    value: event.target.value,
+    goodsList: [...goodsFromServer]
+      .filter(good => good.length >= event.target.value.length),
+  });
+}
 
-export default App;
+render() {
+  return (
+    <div className="App">
+      <h1>Goods</h1>
+      <button
+        type="button"
+        onClick={(event) => {
+          this.hiddenButton(event);
+        }}
+        hidden={this.state.hidden}
+      >
+        Start
+      </button>
+      <button
+        type="button"
+        onClick={() => {
+          this.reverseButton();
+        }}
+        hidden={!this.state.hidden}
+      >
+        Reverse
+      </button>
+      <button
+        type="button"
+        onClick={() => {
+          this.sortABCButton();
+        }}
```

Comment:
```suggestion
        onClick={this.sortABCButton}
```
чому ти не береш до уваги коментарі з попередніх рев'ю?
Я тобі писав використовувати скорочену форму запису якщо ніяких параметрів в колбек прокидувати додаткових не потрібно